packages/google-vertex/src/google-vertex-image-model.ts

```diff
@@ -0,0 +1,84 @@
+import { ImageModelV1, JSONValue } from '@ai-sdk/provider';
+import { Resolvable, resolve } from '@ai-sdk/provider-utils';
+
+export type GoogleVertexImageModelId =
+  | 'imagen-3.0-generate-001'
+  | 'imagen-3.0-fast-generate-001';
+
+interface GoogleVertexImageModelOptions {
+  provider: string;
+  baseURL: string;
+  headers?: Resolvable<Record<string, string | undefined>>;
+  fetch?: typeof fetch;
+}
+
+// https://cloud.google.com/vertex-ai/generative-ai/docs/image/generate-images
+export class GoogleVertexImageModel implements ImageModelV1 {
+  readonly specificationVersion = 'v1';
+
+  get provider(): string {
+    return this.options.provider;
+  }
+
+  constructor(
+    readonly modelId: GoogleVertexImageModelId,
+    private options: GoogleVertexImageModelOptions,
+  ) {}
+
+  async doGenerate(options: {
+    prompt: string;
+    n: number;
+    size: `${number}x${number}` | undefined;
+    providerOptions: Record<string, Record<string, JSONValue>>;
+    abortSignal?: AbortSignal;
+    headers?: Record<string, string>;
+  }): Promise<{ images: string[] }> {
+    const [width, height] = (options.size ?? '1024x1024')
+      .split('x')
+      .map(Number);
```

Comment:
are these only needed for aspect ratio?

Reply:
Yes, removed now, the LLM was basically reverse engineering the aspect ratio from user specified size, and defaulting to the 1:1 imagen defaults to. That's one way to do it, but it's odd and could lead to user surprise since they might not get the size they specified.